lib/plugins/aws/package/compile/events/apiGateway/lib/cors.js

```diff
@@ -11,10 +11,20 @@ module.exports = {
       const corsMethodLogicalId = this.provider.naming
         .getMethodLogicalId(resourceName, 'options');
 
-      // TODO remove once "origins" config is deprecated
       let origin = config.origin;
-      if (config.origins && config.origins.length) {
-        origin = config.origins.join(',');
+      let origins = (config.origins && Array.isArray(config.origins)) ? config.origins : undefined;
+
+      if (origin && origin.indexOf(',') !== -1) {
+        origins = origin.split(',').map(a => a.trim());
```

Comment:
I don't really like the idea of adding extra parsing on top of vanilla(+sls var system) yaml. No real benefit to supporting comma delimited `origin` when `origins` can accept an array.

Reply:
Well, since this changed quite recently so that most modern browsers does only accept one domain in or * as value we need to do this.

“Multiple Values Access-Control-Allow-Origin” by Janosch Maier https://link.medium.com/U8M0LTPHRT

The reason why I suggested to parse both origins and origin is when you want to pass your origins as an env variable or similar (if you have a proxy forward API Gateway method for example) you can't use arrays as env variables